### PR TITLE
Add missing functional tests for Mint

### LINKED ISSUE #156: Missing functional tests for Mint category
- [x] 1. Generate content for any size using stream than creating random bytes
- [x] 2. Missing test for putobject() without content type.
- [x] 3. Missing test for putobject() for unknown sized stream (single/multipart) (may be missing feature)
- [x] 4. Missing test for putobject() for multipart resume (may be missing functionality)
- [x] 5. Missing test for putobject() for client side encryption (may be missing feature)
- [x] 6. Missing test for copyobject() positve/negative tests for modified date
- [x] 7. Missing test for getobject() client side encryption (may be missing feature)
- [x] 8. Missing test for listobject() bucket with objects.
- [x] 9. Missing test for listobject() with prefix and recursive.
- [x] 10. Missing test for listobject() > 1000 objects.
- [x] 11. Missing test for listobject() for version 1 (may be missing feature)
- [x] 12. Missing test removeobject() multiple objects.(may be missing feature)
- [x] 13. Missing test listincompleteupload() for prefix.
- [x] 14. Missing test listincompleteupload() for prefix and recuseive
- [x] 15. Missing test for removeincompleteupload()
- [ ] 16. missing test for presignedgetobject() with default timeout
- [ ] 17. missing test for presignedgetobject() with response headers.
- [ ] 18. missing test for presignedputobject() with default timeout.
- [ ] 19. missing test for thread safey of minio client object.
- [x] 20. categories for all test and quick tests.

### PR DESCRIPTION
**Pull request by poornas**
Fixes #156
- [x]  1.Generate content for any size using stream than creating random bytes _[[Already in place]]_
- [x]  2. Missing test for putobject() without content type.
- [ ]  3. Missing test for putobject() for unknown sized stream (single/multipart) (may be missing feature) _[[will be fixed when #169 is implemented]]_
- [ ]  4. Missing test for putobject() for multipart resume (may be missing functionality) _[[is not part of SDK functionality]]_
- [ ]  5. Missing test for putobject() for client side encryption (may be missing feature) _[[#135 open for missing functionality]]_
- [x]  6. Missing test for copyobject() positve/negative tests for modified date
- [ ]  7. Missing test for getobject() client side encryption (may be missing feature) _[[#135 open for missing functionality]]_
- [x]  8. Missing test for listobject() bucket with objects.
- [x]  9. Missing test for listobject() with prefix and recursive.
- [x]  10. Missing test for listobject() > 1000 objects.
- [ ]  11. Missing test for listobject() for version 1 (may be missing feature) _[[ not implemented in SDK]]_
- [ ]  12. Missing test removeobject() multiple objects.(may be missing feature)_[[ will be fixed by #140]]_
- [x]  13. Missing test listincompleteupload() for prefix.
- [x]  14. Missing test listincompleteupload() for prefix and recursive
- [x]  15. Missing test for removeincompleteupload()
- [ ]  16. missing test for presignedgetobject() with default timeout _[[ will be fixed by #170]]_
- [ ]  17. missing test for presignedgetobject() with response headers_.[[ will be fixed by #170]]_
- [ ]  18. missing test for presignedputobject() with default timeout._[[ will be fixed by #170]]_
- [x]  19. missing test for thread safety of minio client object.
- [x]  20. categories for all test and quick tests.

Also bumping up version and parallelizing put operations in the functional tests.

### CODE REVIEW

**Comment by poornas** [comment]
@krisis  and @balamurugana , this is awaiting your review.

**Comment by balamurugana** [comment]
I see some travis error.  Is it OK to ignore?

File: Minio.Functional.Tests/FunctionalTest.cs [hunk 1]
```diff
@@ -1069,6 +1287,111 @@ public static void Main(string[] args)
...
+                    await minio.PutObjectAsync(bucketName,
+                                                objectName1,
+                                                filestream, filestream.Length, null);
+                }
+                using (MemoryStream filestream = rsg.GenerateStreamFromSeed(1 * MB))
+                {
+                    await minio.PutObjectAsync(bucketName,
+                                                objectName2,
+                                                filestream, filestream.Length, null);
+                }
```
[the remark this answers is not in the document]
Comment (poornas) [reply Minio.Functional.Tests/FunctionalTest.cs]: @krisis, fixed

File: Minio.Functional.Tests/FunctionalTest.cs [hunk 2]
```diff
@@ -1069,6 +1287,111 @@ public static void Main(string[] args)
...
+                ListObjects_Test(minio, bucketName, prefix, 2,true).Wait();
+                System.Threading.Thread.Sleep(5000);
```
[the remark this answers is not in the document]
Comment (poornas) [reply Minio.Functional.Tests/FunctionalTest.cs]: ListObjects returns an IObservable that returns objects as they become available. Though the ListObjects_Test method is declared as returning async Task, the underlying Observable may not have produced all the objects in the bucket, so a sleep allows the list test to complete before the removeobjects call.

**Comment by poornas** [comment]
@balamurugana and @krisis ,  updated the PR to include package bump as this is needed for Mint. Please review whenever possible.

**Comment by poornas** [comment]
@balamurugana , this is waiting on your review.
